### PR TITLE
Various fixes in word cloud (spacing, library update, ...)

### LINKED ISSUE #225: WordCloud Keep Weights Order for Topics
##### Text version

0.2.4.dev

##### Expected behavior
One can observe words in the sorted by their importance according to a specific topic; i.e. sorted by their magnitude regardless of the sign.

##### Actual behavior
WC sorts words by weights hence the most interesting words are at the very top and at the very bottom.

##### Additional info (worksheets, data, screenshots, ...)
What happens with the size? Is it considered correctly for negative weights?

The schema to reproduce:
screen shot 2017-03-31 at 17 21 52

### LINKED ISSUE #390: Word Cloud: don't update on resize
##### Text version

0.5.2

##### Orange version

3.17.dev

##### Expected behavior
Word Cloud stays the same for the same domain.

##### Actual behavior
WC constantly changes. WC should not update on resize. It should also show constant colours for users.

### PR DESCRIPTION
**Pull request by PrimozGodec**
##### Issue

Fixes https://github.com/biolab/orange3-text/issues/399
Fixes #225 
Fixes #390 as much as possible

##### Description of changes
With this PR I made various fixes in the word cloud widget:
- updated the wordcloud.js library
- Addressed extra space in the projection
- Words with smaller weights do not disappear anymore
- The issue from #390 is addressed but not fixed completely. On resize, word cloud must replot, since the distribution of words on canvas must change (otherwise it would look weird). Now words keep the same color after replot.
- Words still overlap and draw one inside each other, this must be fixed in the library. I cannot do any temporary fix.
- Weights that are negative for the topic are now treated as positive in the cloud visualization #225 and colored orange when negative and green when positive (same coloring than in topic widget).
- Moved info box in the status row.
- Code refactoring
- Additional tests

TODO:
- Documentation will be updated (screenshots) when we agree with the changes.

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

File: orangecontrib/text/widgets/owwordcloud.py [hunk 1]
```diff
@@ -342,6 +354,40 @@ def commit(self):
...
+        if (cor_output_len is None
+                and n_selected is None and self.corpus_counter is None):
+            self.info.set_output_summary(self.info.NoOutput)
+        else:
+            cc_len = (
+                len(self.corpus_counter) if self.corpus_counter is not None
+                else 0)
+            input_numbers = f"{cor_output_len or 0}|{n_selected or 0}|{cc_len}"
+            input_string = f"{cor_output_len or 0} documents\n" \
+                           f"{n_selected or 0} selected words" \
```
Comment (ajdapretnar) [inline orangecontrib/text/widgets/owwordcloud.py]: Add line break.

**Comment by ajdapretnar** [comment]
Two things - now the font size is exceedingly big. Font size should be smaller.
Also, font type should be the same as in other Text / Orange widgets. I think this is TNR, but we use... dunno, Lato?

Overlap still occurs. Use election-tweets-2016, Preprocess Text and WC. Perhaps this can't be solved.
Screen Shot 2020-01-09 at 09 29 30

Also, is there any use to 'Regenerate word cloud' button? Does the visualization change at all?

**Comment by ajdapretnar** [comment]
I really appreciate this PR, because WC is mostly a mess. One thing that would be amazing if solved:
Screen Shot 2020-01-09 at 09 31 31
The 'bleeding' / protruding parts of the underlying visualization. See bottom and right side of the visualization. This is beyond annoying.

**Comment by ajdapretnar** [comment]
Comments:
- When I connect Topic Modeling to WC, I get green words for all topic models, even for LDA and HDP, which always give positive words. I don't mind the words being green, but if I deselect 'Color words', words should become gray.
- Sometimes I resize the window to have a nice ellipsoid word cloud. Then I select 'Regenerate word cloud' or even deselect 'Color words' (which prompts replotting) and I get the square version of WC. Very strange.
- Trying to fit all the words for, say, book-excerpts, I need 1/3 of the screen. I think this is a bit excessive. I know we talked about this, but have a look at how this appears on my screen. I have put a couple of widgets alongside for comparison. Perhaps there's no optimal solution here.
Screen Shot 2020-01-13 at 15 57 49
- [IDEA] Could we estimate the number of the words shown in WC by counting the number of characters of all words. Then we make a threshold, say less than 400 chars in WC. 🤔 What do you think?

**Comment by PrimozGodec** [comment]
Thanks for comments:
- Topics are my fault. I forgot that the second two topics have not negative values. So now we have green/red only for the first method and normal colors for others. Also when the first method is used there is special black/white coloring for topics.
- I also realize that behavior when resizing. Resizing is handled by wordcloud2.js itself. It seems that it optimizes resizing and do not recompute all parameters on every resize. It didn't seem too problematic for me since the word cloud is staying nice.  
- I decreased the size of fonts so that the cloud fit in the smaller screen area. I also added the parameter that counts word length combined with their sizes (only word length does not solve the problem when for example many words have high weights). This parameter is now included in word size calculation. The difference is the best visible when we show the topic LDA which gives similar/same values to all words (before all of them were shown with huge fonts). It works on all the examples I have tried. I hope it works on others too.

**Pull request merged**